===================================================================
RCS file: /cvs/gnome/file-roller/src/window.c,v
retrieving revision 1.160
retrieving revision 1.161
diff -u -r1.160 -r1.161
--- window.c	2006/04/03 12:44:13	1.160
+++ window.c	2006/04/04 11:25:47	1.161
@@ -4793,6 +4793,11 @@
 			    junk_paths,
 			    password);
 
+	if (window->archive->process->n_comm < 0) { /* no file to extract */
+		fr_process_start (window->archive->process);
+		return;
+	}
+
 	fr_process_set_continue_func (window->archive->process,
 				      extract__content_is_singleton,
 				      window);
